index.js

```diff
@@ -47,6 +51,57 @@ function fastifyWebsocket (fastify, opts, next) {
   const wss = new WebSocket.Server(wssOptions)
   fastify.decorate('websocketServer', wss)
 
+  async function injectWS (path = '/') {
+    const server2Client = new PassThrough()
+    const client2Server = new PassThrough()
+
+    const serverStream = new Duplexify(server2Client, client2Server)
+    const clientStream = new Duplexify(client2Server, server2Client)
```

Comment:
Did you look at using the [`mock-websocket`](https://www.npmjs.com/package/mock-websocket) library? There's a lot of subtleties and smarts in it for correctly modelling how a websocket is supposed to work. I think there are a few key differences with how the events are emitted, especially on close and terminate where things are carefully choreographed. It'd be good to have the injected websockets model a real one as closely as possible.

Reply:
Can you guide me through it? I looked at it and to me, it's not obvious how to plug it here.